source/AI.cpp

```diff
@@ -2766,9 +2766,44 @@ void AI::DoSurveillance(Ship &ship, Command &command, shared_ptr<Ship> &target)
 		unsigned total = targetShips.size() + targetPlanets.size() + targetSystems.size();
 		if(!total)
 		{
-			// If there is nothing for this ship to scan, have it hold still
-			// instead of drifting away from the system center.
-			Stop(ship, command);
+			// If there is nothing for this ship to scan, have it patrol the entire system
+			// instead of drifting or stopping.
+			// Also allows the ship to land.
+			double radius = ship.GetSystem()->ExtraHyperArrivalDistance();
+
+			// The ship is outside of the effective range of the system,
+			// so we turn it around.
+			if(ship.Position().LengthSquared() > radius * radius)
```

Comment:
can't we just take the length and make sure its positive?

Reply:
Wouldn't that be slower due to sqrt?

Reply:
AI.cpp line 1568:
`else if(ship.Position().Length() >= invisibleFenceRadius)`

Reply:
oh nvm you're right its sqrted I somehow missed it
prob should change this other one too then

anyway you can resolve this